xarray/core/indexes.py

```diff
@@ -1419,6 +1419,11 @@ def check_variables():
         )
 
     indexes = [e[0] for e in elements]
+
+    same_objects = all(indexes[0] is other_idx for other_idx in indexes[1:])
```

Comment:
```suggestion
    indexes_0 = indexes[0]
    same_objects = all(indexes_0 is other_idx for other_idx in indexes[1:])
```
No need to use getitem several times for the same value. Similar thing can be done in other places in the function as well.

Reply:
List indexing is O(1), so should be only a marginal speedup :P

Reply:
Yeah I think the gain in perf would be negligible.